crates/sui-rpc-api/proto/sui.node.v2.proto

```diff
@@ -0,0 +1,363 @@
+  rpc GetCommittee(GetCommitteeRequest) returns (GetCommitteeResponse);
+
+  rpc GetObject(GetObjectRequest) returns (GetObjectResponse);
+  rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse);
+  rpc GetCheckpoint(GetCheckpointRequest) returns (GetCheckpointResponse);
+  rpc GetFullCheckpoint(GetFullCheckpointRequest) returns (GetFullCheckpointResponse);
+
+  //XXX maybe put in another API?
+  rpc ExecuteTransaction(ExecuteTransactionRequest) returns (ExecuteTransactionResponse);
+}
+
+message NodeInfoResponse {
+  // The chain identifier of the chain that this Node is on
+  sui.types.Digest chain_id = 1;
+
+  // Human readable name of the chain that this Node is on
+  string chain = 2;
+
+  // Current epoch of the Node based on its highest executed checkpoint
+  uint64 epoch = 3;
+
+  // Checkpoint height of the most recently executed checkpoint
+  uint64 checkpoint_height = 4;
+
+  // Unix timestamp of the most recently executed checkpoint
+  google.protobuf.Timestamp timestamp = 5;
+
+  // The lowest checkpoint for which checkpoints and transaction data is available
+  optional uint64 lowest_available_checkpoint = 6;
+
+  // The lowest checkpoint for which object data is available
+  optional uint64 lowest_available_checkpoint_objects = 7;
+
+  string software_version = 8;
+}
+
+message HealthCheckRequest {
+  optional uint32 threshold_seconds = 1;
+}
+
+message GetCommitteeRequest {
+  optional uint64 epoch = 1;
+}
+
+message GetCommitteeResponse {
+  sui.types.ValidatorCommittee committee = 1;
+}
+
+message GetTransactionRequest {
+  sui.types.Digest digest = 1;
+
+  optional GetTransactionOptions options = 3;
+}
+
+message GetTransactionOptions {
+  // Include the sui.types.Transaction message in the response.
+  //
+  // Defaults to true if not included
+  optional bool transaction = 1;
+  // Include the Transaction formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool transaction_bcs = 2;
+  // Include the set of sui.types.UserSignature's in the response.
+  //
+  // Defaults to true if not included
+  optional bool signatures = 3;
+  // Include the sui.types.TransactionEffects message in the response.
+  //
+  // Defaults to true if not included
+  optional bool effects = 4;
+  // Include the TransactionEffects formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool effects_bcs = 5;
+  // Include the sui.types.TransactionEvents message in the response.
+  //
+  // Defaults to true if not included
+  optional bool events = 6;
+  // Include the TransactionEvents formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool events_bcs = 7;
+}
+
+message GetTransactionResponse {
+  // The digest of this transaction
+  sui.types.Digest digest = 1;
+  optional sui.types.Transaction transaction = 2;
+  optional sui.types.Bcs transaction_bcs = 3;
+  repeated sui.types.UserSignature signatures = 4;
```

Comment:
So for most top-level core data types we have a bcs format or the protobuf format that you can select between, the only time this wasn't done was for signatures where we only support the protobuf format and not the bcs serialized format. Thoughts here? Should be easy enough to add if we want it though

Reply:
I'd say theres a lot of benefit to having the ability to get back BCS, given how engrained in our stack it is. In particular the protobuf messages can be lossy, in that if you don't understand a field then it just gets dropped, whereas with bcs its fully loss-less. 

There's pros and cons to both of these things are i think have their place depending on the use case. It also sort of maps with how we've exposed many of these things in other apis (jsonrpc, graphql) in that there's a json format and you can also request the raw format